src/Data/Foreign/Class.purs

```diff
@@ -56,6 +56,15 @@ instance arrayIsForeign :: (IsForeign a) => IsForeign (Array a) where
     readElement :: forall a. (IsForeign a) => Int -> Foreign -> F (Array a)
     readElement i value = readWith (ErrorAtIndex i) value
 
+instance eitherIsForeign :: (IsForeign l, IsForeign r) => IsForeign (Either l r) where
+  read value =
+    case (read value) :: F r of
+         Right rval -> pure (Right rval)
```

Comment:
So this prefers the `Right` parse then? I think `Left` would be more common (`aeson` etc.)

Also,  you can probably shorten this using `<|>`.